### ISSUE TITLE
Error when running tests on node 20

### ISSUE THREAD

**Issue by thomas9911** [azu/request-filtering-agent#15]
Im running the tests and I get an internal error:

Node version:

```
$ node --version
v20.8.0
```

OS version:
```
$ cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.1 LTS (Jammy Jellyfish)"
```

Reproduce:

```
yarn test
```

Output:
```
  1) request-filtering-agent
       should not request because the dns-lookuped address is private:
     Uncaught Error: This is caused by either a bug in Node.js or incorrect usage of Node.js internals.
Please open an issue with this stack trace at https://github.com/nodejs/node/issues

      at new NodeError (node:internal/errors:406:5)
      at assert (node:internal/assert:14:11)
      at internalConnect (node:net:1037:3)
      at defaultTriggerAsyncIdScope (node:internal/async_hooks:464:18)
      at GetAddrInfoReqWrap.emitLookup [as callback] (node:net:1481:9)
      at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:130:8)

  2) request-filtering-agent
       "after each" hook in "request-filtering-agent":
     Error: done() called multiple times in hook <request-filtering-agent "after each" hook in "request-filtering-agent"> of file /data/Code/request-filtering-agent/test/request-filtering-agent.test.ts; in addition, done() received error: Error [ERR_INTERNAL_ASSERTION]: This is caused by either a bug in Node.js or incorrect usage of Node.js internals.
Please open an issue with this stack trace at https://github.com/nodejs/node/issues

      at process.emit (node:events:526:35)
      at process.emit (node:domain:488:12)
      at process.emit (node_modules/source-map-support/source-map-support.js:495:21)
      at process._fatalException (node:internal/process/execution:159:25)

  3) request-filtering-agent
       should not request because it is not resolve - timeout:
     Uncaught Error: This is caused by either a bug in Node.js or incorrect usage of Node.js internals.
Please open an issue with this stack trace at https://github.com/nodejs/node/issues

      at new NodeError (node:internal/errors:406:5)
      at assert (node:internal/assert:14:11)
      at internalConnect (node:net:1037:3)
      at defaultTriggerAsyncIdScope (node:internal/async_hooks:464:18)
      at GetAddrInfoReqWrap.emitLookup [as callback] (node:net:1481:9)
      at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:130:8)

  4) request-filtering-agent
       should request public ip address:
     Uncaught Error: This is caused by either a bug in Node.js or incorrect usage of Node.js internals.
Please open an issue with this stack trace at https://github.com/nodejs/node/issues

      at new NodeError (node:internal/errors:406:5)
      at assert (node:internal/assert:14:11)
      at internalConnect (node:net:1037:3)
      at defaultTriggerAsyncIdScope (node:internal/async_hooks:464:18)
      at GetAddrInfoReqWrap.emitLookup [as callback] (node:net:1481:9)
      at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:130:8)



/data/Code/request-filtering-agent/node_modules/mocha/lib/runner.js:962
    throw err;
    ^

Error: This is caused by either a bug in Node.js or incorrect usage of Node.js internals.
Please open an issue with this stack trace at https://github.com/nodejs/node/issues

    at new NodeError (node:internal/errors:406:5)
    at assert (node:internal/assert:14:11)
    at internalConnect (node:net:1037:3)
    at defaultTriggerAsyncIdScope (node:internal/async_hooks:464:18)
    at GetAddrInfoReqWrap.emitLookup [as callback] (node:net:1481:9)
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:130:8) {
  code: 'ERR_INTERNAL_ASSERTION'
}
```

**Comment by azu** [comment]
Thanks for report.

- https://github.com/nodejs/node/issues/48358

Probably, `agent.createConnection = ` override cause this issue.

https://github.com/azu/request-filtering-agent/blob/82efe06a4518dedc1fb96a873eac71bb06214709/src/request-filtering-agent.ts#L127

We need to find workaround this.

**Comment by azu** [comment]
- WIP: https://github.com/azu/request-filtering-agent/pull/16

It seems to work in Node.js 20 by overriding it with a subclass of `http.Agent`.
But, It may includes breaking change. We need to update it as major.(We can not test on Node.js 12, 14).

**Comment by thomas9911** [comment]
Very nice, thank you for the super fast response

**Comment by azu** [comment]
I've released v2.0.0 
https://github.com/azu/request-filtering-agent/releases/tag/v2.0.0
It will work on Node.js 20.
